"Edit" button on code examples

PuruVJ opened the pull request:
Based on #35188, this PR adds the "Edit" button on code examples, clicking which will open up stackblitz editor in new tab with the code example in mind.

Here's a video:

https://user-images.githubusercontent.com/47742487/147886526-f8ed5544-d675-49b4-a6d6-20d211a7fc53.mp4

PuruVJ committed:
feat: add Edit button to open snipets in StackBlitz

PuruVJ committed:
Fix indentation on code examples in Stackblitz

PuruVJ committed:
Update bootstrap script version in script.html

XhmikosR commented:
Hey, thanks for the PR, but this has many unrelated changes.

PuruVJ commented:
Can you specify the unrelated changes? So I can rectify them

XhmikosR commented:
Sorry, I don't have the time to clarify everything. For starters you are not running the lint scripts locally.

Secondly, you are loading the SDK from unpkg. You should use a real CDN and pin the version. And finally, you have hardcoded stuff.

PuruVJ committed:
Use JSdelivr for Stackblitz SDK

PuruVJ committed:
Add EOF

PuruVJ commented:
Changed it from unkpg to jsdelivr

I ran the lint scripts and fixed the issue with EOF on _clipboard-js.scss

If you have time, can you specify what you mean by "hard-coded"?

XhmikosR commented on site/layouts/partials/scripts.html:
```diff
@@ -18,3 +18,39 @@
 <script src="{{ $docsJs.Permalink | relURL }}"></script>
+
+
+{{ printf .Site.Params.cdn.css | safeHTMLAttr }}
```
This is unrelated.

PuruVJ replied:
Fixed in the new PR

XhmikosR commented on site/layouts/partials/scripts.html:
```diff
@@ -18,3 +18,39 @@
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3" crossorigin="anonymous">
```
These are hardcoded

PuruVJ replied:
Fixed in the new PR

XhmikosR committed:
Update example.html

XhmikosR commented on site/layouts/shortcodes/example.html:
```diff
@@ -22,6 +22,9 @@
 
 {{- if eq $show_markup true -}}
+  <div class="bd-edit">
+    <button type="button" class="btn-edit" title="Edit on StackBlitz">Edit</button>
```
It seems to me you are reinventing the wheel with this button's styles.

PuruVJ replied:
Fixed in the new PR

XhmikosR committed:
Update scripts.html

XhmikosR commented on site/assets/js/application.js:
```diff
@@ -4,8 +4,8 @@
  * JavaScript for Bootstrap's docs (https://getbootstrap.com/)
- * Copyright 2011-2021 The Bootstrap Authors
- * Copyright 2011-2021 Twitter, Inc.
+ * Copyright 2011-2022 The Bootstrap Authors
```
Unrelated changes here

PuruVJ replied:
Fixed in the new PR

XhmikosR commented on site/assets/js/application.js:
```diff
@@ -169,4 +169,43 @@
   anchors.add('.bd-content > h2, .bd-content > h3, .bd-content > h4, .bd-content > h5')
+
+  // Open in StackBlitz logic
+  /*
```
No need to keep commented out code

PuruVJ replied:
Fixed in Original PE

XhmikosR committed:
Update application.js

XhmikosR commented on site/layouts/partials/scripts.html:
```diff
@@ -18,3 +18,39 @@
+
+{{ printf .Site.Params.cdn.css | safeHTMLAttr }}
+
+<script src="https://cdn.jsdelivr.net/npm/@stackblitz/sdk@1.5.3/bundles/sdk.umd.min.js"></script>
```
We don't need to load this in every page. Which probably means, you shouldn't add the logic in application.js.

PuruVJ replied:
Which file do you suggest I add this logic?

PuruVJ replied:
Would adding a script tag to the `example.html` partial and putting "open in stackblitz" logic there be fine?

PuruVJ commented:
Hi, I will close this PR and make a new one with new changes. I am having some issues in git locally

PuruVJ commented:
New PR here #35644
